### ISSUE TITLE
Vulkan renderer broken in ROBLOX Player / Studio

### ISSUE THREAD

**Issue by imskyyc** [Frogging-Family/wine-tkg-git#797]
Recently, since the later versions of Staging 7.11, the ROBLOX Player & Studio have not worked with using forced Vulkan renderer, and good performance is not really achievable through DXVK

The player doesn't even output any logs.

Studio error log:
[2022-07-03_17-13-12_RobloxStudioLauncherBeta.exe_stderr.log](https://github.com/Frogging-Family/wine-tkg-git/files/9035719/2022-07-03_17-13-12_RobloxStudioLauncherBeta.exe_stderr.log)

Note: this was done using RADV drivers, and not AMDVLK. There is no difference however.
.

**Comment by imskyyc** [comment]
Right, sorry. Forgot to mention that you basically have to run it through Grapejuice, which is a sort of wrapper for Roblox client stuff that allows it to function properly. https://brinkervii.gitlab.io/grapejuice/docs/

The latest compiled wine-tkg from staging 7.12 works fine in terms of the installer / launching.

**Comment by imskyyc** [comment]
Ammending the previous, apparently it doesn't work with the latest tkg-staging 7.12, but it did yesterday. Reverting to tkg-staging 7.11.r6 fixes the launching issue. Vulkan renderer issues persist.

This is what the player logs out.
[2022-07-04_04-58-36_RobloxPlayerLauncher.exe_stderr.log](https://github.com/Frogging-Family/wine-tkg-git/files/9039190/2022-07-04_04-58-36_RobloxPlayerLauncher.exe_stderr.log)

**Comment by Tk-Glitch** [comment]
I am using Grapejuice.
Sadly I tested builds ranging from 7.0 to 7.12 (staging, mainline, proton) with the exact same crash every time. I have never attempted to run the game ever before btw.
From your log I notice that it complains about no vk childwindow support, which would require either `_proton_fs_hack="true"` or `_childwindow_fix="true"`. If one of those is enabled in your builds and the log is unchanged, then we have an issue on our end indeed.

**Comment by imskyyc** [comment]
Just checked, childwindow fix is enabled. I'll try compiling against the latest staging 7.11 to see if that changes anything, but I doubt it will.

**Comment by Tk-Glitch** [comment]
You could try reverting `https://github.com/Frogging-Family/wine-tkg-git/commit/7ea035c19d63b81f819330af44f8d5bff2cac7ba`. The change can break sensitive apps, and this might be the first report of it breaking :D

**Comment by imskyyc** [comment]
Trying it now, I ran "git revert 7ea035c19d63b81f819330af44f8d5bff2cac7ba", that should work right? I have zero clue how to mess with git reverts, patches, etc lol

**Comment by Tk-Glitch** [comment]
> Trying it now, I ran "git revert [7ea035c](https://github.com/Frogging-Family/wine-tkg-git/commit/7ea035c19d63b81f819330af44f8d5bff2cac7ba)", that should work right? I have zero clue how to mess with git reverts, patches, etc lol

Yeah that should work :)

**Comment by imskyyc** [comment]
No dice, studio abnormally terminates, and the player has the same issue.

**Comment by Tk-Glitch** [comment]
Looks like an upstream regression then. Unfortunately I'll need to get the game to install first before I can get any bisecting done :/

**Comment by imskyyc** [comment]
So I'm not sure if this was some weird fluke, but all of a sudden today's release (7.12.r1) started working with RADV drivers. Then I tried installing AMDVLK due to horrid performance, and it hasn't worked since. It was likely some weird thing with Grapejuice, since I had performance similar to if I was using DX11 or OpenGL. Just thought I'd put that here.

**Comment by imskyyc** [comment]
As of Wine 7.12.r6, the Vulkan issue in the player has been fixed on AMDVLK, no RADV support to be seen anywhere. Studio is still broken on all Vulkan renderers.

**Comment by imskyyc** [comment]
As of a recent ROBLOX update, Studio no longer works at all. Player continues to run fine, with lots of stuttering after playing for a long time.

**Comment by imskyyc** [comment]
I have found through testing that Studio runs perfectly fine on wine-staging 7.13, but not on wine-tkg 7.13. Here's my most recent stderr from Studio. Player still runs fine, as well as with RADV.
[2022-07-23_12-04-48_RobloxStudioLauncherBeta.exe_stderr.log](https://github.com/Frogging-Family/wine-tkg-git/files/9174462/2022-07-23_12-04-48_RobloxStudioLauncherBeta.exe_stderr.log)

**Comment by Tk-Glitch** [comment]
I'm seeing `vulkan:X11DRV_vkCreateWin32SurfaceKHR Application requires child window rendering, which is not implemented yet!` in that log, which is indeed a feature that's not implemented in staging, but for which we offer a patch. I assume disabling vk childwindow, which is enabled in our default config (and the fshack in case it's enabled, as it also contains vk childwindow support) might make it work if the game runs fine on plain staging.

**Comment by imskyyc** [comment]
Not sure how it happened, but as it turns out, it was defaulting to tkg 7.7, which did work with Studio. Not sure what got changed from then to now.

**Comment by ColdIce1605** [comment]
ok, not sure if this is of note but I've also have this issue it seems to only happen after the prefix in grapejuice for studio is created(or studio is installed whatever it is) this specifically happens when editting FFLAGS
